### DISCUSSION TITLE
Question for whatsapp blocking

### DISCUSSION THREAD

**Discussion by travelg** [blocklistproject/Lists#303] [Q&A] [answered]
Hello, Thanks for nice whatspp list. I have added whatsapp.txt file in my pihole but it do not block the messages and whatsapp call.

Please tell me what I am doing wrong? I have added https://github.com/blocklistproject/Lists/blob/master/whatsapp.txt 
in the adlist in pihole. It is blocking the images download from any whatsapp messages but I can still send messages and receive it. 

I want to just block completely whatsapp in my network. 
thanks for you help.

**Comment by CYB3RL4MBD4** [comment]
Maybe try to use this link instead?

[https://blocklistproject.github.io/Lists/whatsapp.txt](https://blocklistproject.github.io/Lists/whatsapp.txt)

And don't forget to launch "Pi-hole -g" command to update the database and add the domains to block.

**Comment by travelg** [comment]
@CYB3RL4MBD4 Thanks this helped. So now Whatsapp website is blocked in browser and whatsapp media is not able to download But
Still I am able to send and receive messages. 
I want to also block whatsapp messages send and receive. 
Do you have any solution ? thanks.

**Reply by blocklistproject** [reply]
Made a pretty big update to the Whatsapp list. Check and see if blocking is working now.

**Comment by CYB3RL4MBD4** [comment]
> @CYB3RL4MBD4 Thanks this helped. So now Whatsapp website is blocked in browser and whatsapp media is not able to download
> 
You're welcome.
> But still I am able to send and receive messages. 
> 
> I want to also block whatsapp messages send and receive. 
> 
> Do you have any solution ? thanks.
>
That's weird but it could be that some URL's specific to message sending/receiving are not on the list. Is this only happening with the phone app?

**Comment by travelg** [comment]
> > @CYB3RL4MBD4 Thanks this helped. So now Whatsapp website is blocked in browser and whatsapp media is not able to download
> 
> You're welcome.
> 
> > But still I am able to send and receive messages.
> > I want to also block whatsapp messages send and receive.
> > Do you have any solution ? thanks.
> 
> That's weird but it could be that some URL's specific to message sending/receiving are not on the list. Is this only happening with the phone app?

Yes. It happens with phone whatsapp app. Do you know which url is missing to block whatsapp messages and call thanks.

**Comment by Trainax** [comment]
Have you tried rebooting your phone? Maybe the domain is cached in your phone from when it wasn't blocked yet

> 
> 
> > > @CYB3RL4MBD4 Thanks this helped. So now Whatsapp website is blocked in browser and whatsapp media is not able to download
> > 
> > 
> > You're welcome.
> > > But still I am able to send and receive messages.
> > > I want to also block whatsapp messages send and receive.
> > > Do you have any solution ? thanks.
> > 
> > 
> > That's weird but it could be that some URL's specific to message sending/receiving are not on the list. Is this only happening with the phone app?
> 
> Yes. It happens with phone whatsapp app. Do you know which url is missing to block whatsapp messages and call thanks.

Have you tried rebooting your phone? Maybe the domain is cached in your phone from when it wasn't blocked yet

**Comment by CYB3RL4MBD4** [comment]
Good point! Thanks!

**Comment by CYB3RL4MBD4** [comment]
Also maybe flush Pi-hole dns cache, this can also be the cause of some domains continuing to work.

**Comment by travelg** [comment]
> Also maybe flush Pi-hole dns cache, this can also be the cause of some domains continuing to work.

Yes. I tried restarting my phone cleaning dns cache. And also trying after 12 hours same issue. Messages are still able to send.  Do you have some solution? Thanks.

**Comment by CYB3RL4MBD4** [comment]
Did you try to display live log of your pi-hole, and initiate a WhatsApp message? You should see the incriminated domain appear in the log. Sorry personally I don't use WhatsApp, I cannot try.

**Comment by Trainax** [comment]
> Did you try to display live log of your pi-hole, and initiate a WhatsApp message? You should see the incriminated domain appear in the log. Sorry personally I don't use WhatsApp, I cannot try.

I tried rebooting my phone and looking at the domains. The domains which get called are:

- privatestats.whatsapp.net (blocked by StevenBlack's hosts list)
- g.whatsapp.net (which is on the WhatsApp list)
- static.whatsapp.net
- mmg.whatsapp.net (which is on the WhatsApp list)

I would try blocking static.whatsapp.net, rebooting the phone to flush the dns cache and see if this works. Unfortunately I can't try myself because WhatsApp is needed in my network

**Comment by travelg** [comment, marked as answer]
> > Did you try to display live log of your pi-hole, and initiate a WhatsApp message? You should see the incriminated domain appear in the log. Sorry personally I don't use WhatsApp, I cannot try.
> 
> I tried rebooting my phone and looking at the domains. The domains which get called are:
> 
> * privatestats.whatsapp.net (blocked by StevenBlack's hosts list)
> * g.whatsapp.net (which is on the WhatsApp list)
> * static.whatsapp.net
> * mmg.whatsapp.net (which is on the WhatsApp list)
> 
> I would try blocking static.whatsapp.net, rebooting the phone to flush the dns cache and see if this works. Unfortunately I can't try myself because WhatsApp is needed in my network

Thanks for information. I have already restarted phone and cleared dns. I have already blocked with regex all whatsapp.net domains. Still messages can be sent and received.

**Reply by Trainax** [reply]
> I have already blocked with regex all whatsapp.net domains

Have you also blocked all whatsapp.com domains with regex?

**Reply by travelg** [reply]
yes. i will test it again tomorrow and I will comment here.

**Reply by travelg** [reply]
now it works. thanks.
